### PR TITLE
Training an RNN to classify sequences using just the last LSTM output

### PR DESCRIPTION
**Pull request by aodhan-domhnaill**
I did this as a demo for a particular project. I can polish it up if you want it.

You basically just run an LSTM on every sequence, take the last output for each sequence and the push that through a neural net.

There are 2 demos training on generated data and then one classifying sentences as Spanish or English. The language one is tested using entirely different novels.

### CODE REVIEW

**Comment by zackchase** [comment]
Hi aidan - just got back to town after a couple weeks abroad plus some insane paper deadline - looking forward to reviewing this properly. Thanks for the PR!

**Review by ChaiBapchya — COMMENTED** [review]
What kind of loss is being used here -`loss = nd.mean((out[out.shape[0]-1] - 2)**2)`
? Would be great to mention that.

File: chapter05_recurrent-neural-networks/lstm-last-output.ipynb [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    "        (0, 'http://www.gutenberg.org/cache/epub/10822/pg10822.txt'), # Spanish\n",
+    "        (1, 'https://www.gutenberg.org/files/1342/1342-0.txt'), # English\n",
+    "    ]\n",
+    "    \n",
+    "    def sentences(novel, min_len=30, max_len=300):\n",
+    "        for s in re.sub(r'\\s+', ' ', requests.get(novel).text).split('.')[20:520]:\n",
+    "            if len(s) > min_len and len(s) < max_len:\n",
+    "                yield s\n",
+    "    \n",
+    "    # Run through all the novels and grab all the sentences in the books and grab all the sentences\n",
```
Comment (ChaiBapchya) [inline chapter05_recurrent-neural-networks/lstm-last-output.ipynb:318]: What do you mean here? @aidan-plenert-macdonald

**Review by ChaiBapchya — APPROVED** [review]
LGTM for the most part. Minor comments!

File: chapter05_recurrent-neural-networks/lstm-last-output.ipynb (hunk 1 above)
Comment (aodhan-domhnaill) [reply chapter05_recurrent-neural-networks/lstm-last-output.ipynb:318]: Just sentences out as series of characters
